lib/ansible/module_utils/docker_common.py

```diff
@@ -381,7 +382,7 @@ def find_image(self, name, tag):
 
         self.log("Find image %s:%s" % (name, tag))
         images = self._image_lookup(name, tag)
-        if len(images) == 0:
+        if len(images) == 0 and name not in DOCKER_RESERVED_IMAGE_NAMES:
             # In API <= 1.20 seeing 'docker.io/<name>' as the name of images pulled from docker hub
```

Comment:
The problem with unpicking the usage of `find_image` is that its used in 1 or 2 places in `docker_container` and *everywhere* in `docker_image_facts` and `docker_image` which makes this rather a large change, unless the interface just gets bifurcated for the time being between "things which deal with repository:tag and things which deal with a name"